tests/integration/test_hostprovider.py

```diff
@@ -0,0 +1,111 @@
+    hostprovider = await adcm_client.hostproviders.create(
+        bundle=hostprovider_bundle, name="Hostprovider name", description="Hostprovider description"
+    )
+    assert hostprovider.display_name == "complex_provider"
+    assert hostprovider.name == "Hostprovider name"
+    assert hostprovider.description == "Hostprovider description"
+    assert isinstance(hostprovider.actions, ActionsAccessor)
+    assert isinstance(await hostprovider.config, ObjectConfig)
+    assert isinstance(hostprovider.config_history, ConfigHistoryNode)
+    assert isinstance(hostprovider.upgrades, UpgradeNode)
+    hosts = await hostprovider.hosts.all()
+    assert len(hosts) == 0
+
+
+async def _test_hostprovider_accessors(adcm_client: ADCMClient, hostprovider_bundle: Bundle) -> None:
+    for new_host_provider in ["hostprovider-1", "hostprovider-2", "hostprovider-3"]:
+        await adcm_client.hostproviders.create(
+            bundle=hostprovider_bundle, name=new_host_provider, description=new_host_provider
+        )
+
+    hostprovider = await adcm_client.hostproviders.get(name__eq="hostprovider-1")
+    assert isinstance(hostprovider, HostProvider)
+    assert hostprovider.name == "hostprovider-1"
+
+    with pytest.raises(ObjectDoesNotExistError):
+        await adcm_client.hostproviders.get(name__eq="fake_hostprovider")
+
+    with pytest.raises(MultipleObjectsReturnedError):
+        await adcm_client.hostproviders.get(name__icontains="pr")
+
+    assert not await adcm_client.hostproviders.get_or_none(name__eq="fake_hostprovider")
+    assert isinstance(await adcm_client.hostproviders.get_or_none(name__contains="hostprovider-1"), HostProvider)
+
+    assert len(await adcm_client.hostproviders.all()) == len(await adcm_client.hostproviders.list()) == 4
+
+    hostproviders_list = await adcm_client.hostproviders.list(query={"limit": 2, "offset": 1})
+    assert isinstance(hostproviders_list, list)
+    assert len(hostproviders_list) == 2
+
+    hostproviders_list = await adcm_client.hostproviders.list(query={"offset": 4})
+    assert isinstance(hostproviders_list, list)
+    assert len(hostproviders_list) == 0
+
+    async for hp in adcm_client.hostproviders.iter():
+        assert isinstance(hp, HostProvider)
+        assert "hostprovider" in hp.name.lower()
+
+    assert len(await adcm_client.hostproviders.filter(bundle__eq=hostprovider_bundle)) == 4
+
+    await hostprovider.delete()
+
+
+async def _test_pagination(adcm_client: ADCMClient, tmp_path: Path) -> None:
+    created_bundles = await create_bundles_by_template(
+        adcm_client,
+        tmp_path,
+        BUNDLES / "simple_hostprovider",
+        target_name="name",
+        field_to_modify="simple_provider",
+        new_value="provider_name",
+        number_of_bundles=55,
+    )
+    for i, bundle in enumerate(created_bundles):
+        await adcm_client.hostproviders.create(bundle=bundle, name=f"Hostprovider name {i}")
```

Comment:
no need in 50 bindles in here, just 50 HPs 

but generally the idea to create HPs from different bundles is good, because we should check that filter by bundle works

Reply:
removed